mautic/mautic
# Campaign events with a 0 delay ignore configured days of week

| Q                                      | A
| -------------------------------------- | ---
| Branch  | staging
| Bug fix  | yes
| New feature? | no
| Deprecations? | no
| BC breaks  | no
| Automated tests included? | yes
| Related user documentation PR URL | NA
| Related developer documentation PR URL | NA
| Issue(s) addressed | 

#### Description:
Campaign events with a 0 delay would ignore the day of week restrictions.

#### Steps to test this PR:
1. Load up this pr
2. Create a 0 delay event that restricts the days so that only the previous day of week is allowed.
3. When saving the event should not fire immediately

#### Other areas of Mautic that may be affected by the change:
None

#### List deprecations along with the new alternative:
No deprecations have been added

[//]: # ( As applicable: )
#### List of areas covered by the unit and/or functional tests:
1. Added many unit test and a few functional test.

<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/11292"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
